modules/phylowgs/1.0/phylowgs.smk

```diff
@@ -0,0 +1,391 @@
+    input:
+        maf = str(rules._phylowgs_input_maf.output.maf),
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        vcf = temp(CFG['dirs']['maf_to_vcf'] + "{seq_type}--{genome_build}/{patient_id}/{tumour_id}--{normal_id}--{pair_status}.maf_to.vcf")
+    conda:
+        CFG["conda_envs"]["vcf2maf"]
+    group: "input_maf"
+    shell:
+        op.as_one_line("""
+        maf2vcf.pl --input-maf {input.maf} --output-dir $(dirname {output.vcf}) --output-vcf {output.vcf} --ref-fasta {input.fasta}
+        """)
+
+rule _phylowgs_priority_ssms:
+    input:
+        mafs = get_input_mafs
+    output:
+        ssms = CFG['dirs']['maf_to_vcf'] + "{seq_type}--{genome_build}/{patient_id}/coding_ssms.txt"
+    params:
+        noncoding = CFG["scripts"]["noncoding"]
+    conda:
+        CFG["conda_envs"]["coreutils"]
+    shell:
+        op.as_one_line("""
+        grep -hvf {params.noncoding} {input.mafs} | awk '{{FS="\\t"}} {{OFS="_"}} {{print $5, $6}}' | sed 's/chr//g' > {output.ssms}
+        """)
+
+rule _phylowgs_bgzip_vcf:
+    input:
+        vcf = str(rules._phylowgs_maf_to_vcf.output.vcf)
+    output:
+        vcf = temp(CFG['dirs']['maf_to_vcf'] + "{seq_type}--{genome_build}/{patient_id}/{tumour_id}--{normal_id}--{pair_status}.maf_to.vcf.gz"),
+        tbi = CFG['dirs']['maf_to_vcf'] + "{seq_type}--{genome_build}/{patient_id}/{tumour_id}--{normal_id}--{pair_status}.maf_to.vcf.gz.tbi",
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    shell:
+        op.as_one_line("""
+        bcftools sort {input.vcf} | bcftools view -s "{wildcards.normal_id},{wildcards.tumour_id}" -i 'FMT/DP[0] > 0 && FMT/AD[0:1] > 1' -Oz -o {output.vcf} && tabix -p vcf {output.vcf}
+        """)
+
+
+# Preprocess vcf and battenberg inputs together
+rule _phylowgs_create_inputs:
+    input:
+        create_phylowgs_inputs,
+        priority = str(rules._phylowgs_priority_ssms.output.ssms)
+    output:
+        ssms = CFG["dirs"]["preprocess_inputs"] + "{seq_type}--{genome_build}/{patient_id}/ssm_data.txt",
+        cnvs = CFG["dirs"]["preprocess_inputs"] + "{seq_type}--{genome_build}/{patient_id}/cnv_data.txt",
+        params = CFG["dirs"]["preprocess_inputs"] + "{seq_type}--{genome_build}/{patient_id}/params.json"
+    log:
+        stderr = CFG["logs"]["preprocess_inputs"] + "{seq_type}--{genome_build}/{patient_id}/create_phylowgs_inputs.stderr.log",
+        stdout = CFG["logs"]["preprocess_inputs"] + "{seq_type}--{genome_build}/{patient_id}/create_phylowgs_inputs.stdout.log"
+    params:
+        cli = create_phylowgs_inputs_cli,
+        opts = CFG["options"]["create_inputs"]["opts"],
+        sex = lambda w: config["lcr-modules"]["phylowgs"]["switches"]["sex"][op.filter_samples(PATIENTS, tumour_patient_id = w.patient_id)["tumour_sex"].values[0]] if op.filter_samples(PATIENTS, tumour_patient_id = w.patient_id)["tumour_sex"].values[0] in config["lcr-modules"]["phylowgs"]["switches"]["sex"].keys() else "auto",
```

Comment:
Do you think this can be a separate designated function?

Reply:
It's a bit verbose but it's IMO a good use of a lambda function.